pkg/serviceprovider/access_token_mapper.go

```diff
@@ -54,23 +58,27 @@ func (at AccessTokenMapper) ToSecretType(secretType corev1.SecretType, mapping *
 	case corev1.SecretTypeDockercfg:
 		ret[corev1.DockerConfigKey] = at.Token
 	case corev1.SecretTypeDockerConfigJson:
-		// parse host from ServiceProviderUrl if that is not possible use the whole ServiceProviderUrl as host
-		host := at.ServiceProviderUrl
 		parsed, err := url.Parse(at.ServiceProviderUrl)
-		if err == nil && parsed.Host != "" {
-			host = parsed.Host
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse service provider url: %w", err)
+		}
+		host := parsed.Host
+		if host == "" {
+			host = at.ServiceProviderUrl
```

Comment:
I don't undestand this. In what cases it can happen and why we're writing `ServiceProviderUrl` to host?